spec/lib/spec/lib/user_comm_screener_spec.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe UserCommScreener do
+  fab!(:target_user1) { Fabricate(:user, username: "bobscreen") }
+  fab!(:target_user2) { Fabricate(:user, username: "hughscreen") }
+  fab!(:target_user3) { Fabricate(:user, username: "alicescreen") }
+  fab!(:target_user4) { Fabricate(:user, username: "janescreen") }
+  fab!(:target_user5) { Fabricate(:user, username: "maryscreen") }
+
+  before do
+    MutedUser.create!(user: target_user1, muted_user: acting_user)
+    IgnoredUser.create!(user: target_user2, ignored_user: acting_user, expiring_at: 2.days.from_now)
+    target_user3.user_option.update(allow_private_messages: false)
+  end
+
+  subject { described_class.new(
+    acting_user, target_usernames: ["bobscreen", "hughscreen", "alicescreen", "janescreen", "maryscreen"])
+  }
+
+  context "when the actor is not staff" do
+    fab!(:acting_user) { Fabricate(:user) }
+
+    describe "allowing_actor_communication" do
+      it "returns the usernames of people not ignoring, muting, or disallowing PMs from the actor" do
+        expect(subject.allowing_actor_communication).to eq(["janescreen", "maryscreen"])
+      end
+    end
+
+    describe "preventing_actor_communication" do
+      it "returns the usernames of people ignoring, muting, or disallowing PMs from the actor" do
+        expect(subject.preventing_actor_communication).to eq(["bobscreen", "hughscreen", "alicescreen"])
+      end
+    end
+
+    describe "ignoring_or_muting_actor?" do
+      it "does not raise an error when looking for a user who has no communication preferences" do
```

Comment:
If we're expecting it to not raise an error I think the better assert is `.not_to raise_error`.